### PR TITLE
MON-16653 Add extra delivery to artifactory

### PR DESCRIPTION
**Pull request by tuntoja**
## Description

**PLEASE MAKE SURE THAT THE BRANCH PR INCLUDES JIRA TICKET ID** (_for centreon-internal_)

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [x] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/actions/deb-delivery/action.yml [hunk 1]
```diff
@@ -25,6 +25,9 @@ inputs:
   stability:
     description: "The package stability (stable, testing, unstable)"
     required: true
+  arti_token:
+    description: "test token for arti"
```
Comment (kduret) [inline .github/actions/deb-delivery/action.yml:29]:
```suggestion
    description: "token for artifactory"
```

File: .github/actions/deb-delivery/action.yml [hunk 2]
```diff
@@ -25,6 +25,9 @@ inputs:
   stability:
     description: "The package stability (stable, testing, unstable)"
     required: true
+  arti_token:
```
Comment (kduret) [inline .github/actions/deb-delivery/action.yml:28]: IMHO it is more meaningful with full name
```suggestion
  artifactory_token:
```

File: .github/actions/deb-delivery/action.yml [hunk 3]
```diff
@@ -54,6 +57,7 @@ runs:
             curl -v -u "${{ inputs.nexus_username }}":"${{ inputs.nexus_password }}" -H "Content-Type: multipart/form-data" --data-binary "@./$FILE" https://apt.centreon.com/repository/23.04/
           else
             curl -v -u "${{ inputs.nexus_username }}":"${{ inputs.nexus_password }}" -H "Content-Type: multipart/form-data" --data-binary "@./$FILE" https://apt.centreon.com/repository/23.04-${{ inputs.stability }}/
+            curl -v -H "Authorization: Bearer ${{ inputs.arti_token }}" -X PUT "https://centreon.jfrog.io/artifactory/apt-${{ inputs.version }}-${{ inputs.stability }}/pool/$FILE;deb.distribution=bullseye;deb.component=main;deb.architecture=amd64" -T "$FILE"
```
Comment (kduret) [inline .github/actions/deb-delivery/action.yml:60]:
```suggestion
            curl -v -H "Authorization: Bearer ${{ inputs.artifactory_token }}" -X PUT "https://centreon.jfrog.io/artifactory/apt-${{ inputs.version }}-${{ inputs.stability }}/pool/$FILE;deb.distribution=bullseye;deb.component=main;deb.architecture=amd64" -T "$FILE"
```

File: .github/actions/rpm-delivery/action.yml [hunk 4]
```diff
@@ -46,6 +46,9 @@ inputs:
   stability:
     description: "The package stability (stable, testing, unstable)"
     required: true
+  arti_token:
```
Comment (kduret) [inline .github/actions/rpm-delivery/action.yml:49]:
```suggestion
  artifactory_token:
```

File: .github/actions/rpm-delivery/action.yml [hunk 5]
```diff
@@ -46,6 +46,9 @@ inputs:
...
+    description: "test token for arti"
```
Comment (kduret) [inline .github/actions/rpm-delivery/action.yml:50]:
```suggestion
    description: "token for artifactory"
```

File: .github/workflows/awie.yml [hunk 6]
```diff
@@ -87,6 +87,7 @@ jobs:
           release: ${{ needs.get-version.outputs.release }}
           nexus_username: ${{ secrets.ARTI_USER }}
           nexus_password: ${{ secrets.ARTI_PASSWD }}
+          arti_token: ${{ secrets.ARTI_APT_TEST_TOKEN }}
```
Comment (kduret) [inline .github/workflows/awie.yml:90]: same suggestion for others :)
```suggestion
          artifactory_token: ${{ secrets.ARTI_APT_TEST_TOKEN }}
```

File: .github/actions/rpm-delivery/action.yml [hunk 7]
```diff
@@ -95,10 +98,12 @@ runs:
           # Delivery with proper repository structure according to stability
           if [ "${{ inputs.stability }}" == stable ]; then
             TARGET="/srv/centreon-yum/yum.centreon.com/$PROJECT_PATH/$MAJOR/$DISTRIB/$REPOTYPE/$ARCH/RPMS"
+            curl -v -H "Authorization: Bearer ${{ inputs.arti_token }}" -X PUT "https://centreon.jfrog.io/artifactory/rpm/$PROJECT_PATH/$MAJOR/$DISTRIB/${{ inputs.stability }}/$ARCH/$FILE" -T "./$FILE"
           else
             FOLDER="centreon-$PROJECT-$MAJOR.$MINOR-$RELEASE"
             TARGET="/srv/centreon-yum/yum.centreon.com/$PROJECT_PATH/$MAJOR/$DISTRIB/$REPOTYPE/$ARCH/$PROJECT/$FOLDER"
             PROJECT_LOCATION="/srv/centreon-yum/yum.centreon.com/$PROJECT_PATH/$MAJOR/$DISTRIB/$REPOTYPE/$ARCH/$PROJECT"
+            curl -v -H "Authorization: Bearer ${{ inputs.arti_token }}" -X PUT "https://centreon.jfrog.io/artifactory/rpm/$PROJECT_PATH/$MAJOR/$DISTRIB/${{ inputs.stability }}/$ARCH/$PROJECT/$FOLDER/$FILE" -T "./$FILE"
```
Comment (kduret) [inline .github/actions/rpm-delivery/action.yml:106]: construct path and make a single curl call after `if..else`

File: .github/workflows/get-version.yml [hunk 8]
```diff
@@ -86,7 +86,7 @@ jobs:
           esac
 
           case "$BRANCHNAME" in
-            develop | dev-[2-9][0-9].[0-9][0-9].x)
+            develop | dev-[2-9][0-9].[0-9][0-9].x | MON-16653-trial-saa-s-artifactory)
```
Comment (kduret) [inline .github/workflows/get-version.yml:89]: do not forget to remove this once everything is ready

File: .github/actions/deb-delivery/action.yml (hunk 3 above)
Comment (kduret) [inline .github/actions/deb-delivery/action.yml:60]: artchitecture is not always amd64

Comment (kduret) [reply .github/actions/deb-delivery/action.yml]: will be managed in an other PR

File: .github/actions/rpm-delivery/action.yml (hunk 7 above)
Comment (kduret) [reply .github/actions/rpm-delivery/action.yml:106]: will be managed in an other PR

**Review by kduret — APPROVED** [review]

**Review by paloth — APPROVED** [review]

**Pull request merged**